website/content/docs/concepts/security/data-encryption.mdx

```diff
@@ -55,6 +58,53 @@ provided in this section is purely for informational purposes.
 - `sessions`: This is used as a base key against which to derive
   session-specific encryption keys.
 
+### Key version lifecycle Management
+
+The lifecycle of the per-scope KEK and DEKs can be controlled via the CLI or
+API using the key endpoints under the scopes section. To rotate all the keys
+in a scope, use the `rotate-keys` endpoint:
+
+```shell-session
+$ boundary scopes rotate-keys -scope-id p_A4jfDjZ9jf
+```
+
+This creates new key versions for all keys in the scope `p_A4jfDjZ9jf`, and
+makes these key versions the active key versions for encrypting new data. The
+previous key version(s) will continue to be used for decrypting existing data.
```

Comment:
I'm sure that has been talked about in the past, but is that expected behavior from users to only encrypt new data with the new key after you do a rotation? Was it mostly a time consideration not needing to re-encrypt all of the existing data?

Reply:
My understand is that this sort of process is pretty standard - when rotating a key, it doesn't actually re-encrypt all existing data, it just means new data uses the new key. Re-encrypting all existing data could be a very expensive process. In the future we may do it in the background to automatically destroy key versions that are old, but we're not doing it right now, and I don't think we'll ever do it synchronously with a key rotation. With that in mind, do you think this sentence still makes sense?